// @kintone/plugin-uploader/dist/params.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.inquireParams = void 0;
const inquirer = __importStar(require("inquirer"));
const messages_1 = require("./messages");
const inquireParams = (_a) => __awaiter(void 0, [_a], void 0, function* ({ username, password, baseUrl, lang, }) {
    const m = (0, messages_1.getBoundMessage)(lang);
    const questions = [
        {
            type: "input",
            message: m("Q_BaseUrl"),
            name: "baseUrl",
            default: baseUrl,
            when: () => !baseUrl,
            validate: (v) => !!v,
        },
        {
            type: "input",
            name: "username",
            message: m("Q_UserName"),
            default: username,
            when: () => !username,
            validate: (v) => !!v,
        },
        {
            type: "password",
            name: "password",
            message: m("Q_Password"),
            default: password,
            when: () => !password,
            validate: (v) => !!v,
        },
    ];
    return inquirer.prompt(questions).then((answers) => {
        var _a, _b, _c;
        return ({
            username: (_a = answers.username) !== null && _a !== void 0 ? _a : username,
            password: (_b = answers.password) !== null && _b !== void 0 ? _b : password,
            baseUrl: (_c = answers.baseUrl) !== null && _c !== void 0 ? _c : baseUrl,
        });
    });
});
exports.inquireParams = inquireParams;
